packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -114,7 +116,11 @@ export function executeStepDependencyGraph<
   const workingGraph = inputGraph.clone();
 
   // create a queue for managing promises to be executed
-  const promiseQueue = new PromiseQueue();
+  const promiseQueueOptions: { concurrency?: number } = {};
+  if (stepConcurrency !== undefined) {
+    promiseQueueOptions.concurrency = stepConcurrency;
+  }
+  const promiseQueue = new PromiseQueue(promiseQueueOptions);
```

Comment:
PromiseQueue does not allow you to send `{ concurrency: undefined }` so we only put it on the options if we have a value.